pkg/volume/csi/csi_block.go

```diff
@@ -252,22 +252,66 @@ func (m *csiBlockMapper) SetUpDevice() (string, error) {
 	}
 
 	// Call NodeStageVolume
-	stagingPath, err := m.stageVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment)
+	_, err = m.stageVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment)
 	if err != nil {
 		return "", err
 	}
 
-	// Call NodePublishVolume
-	publishPath, err := m.publishVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment, stagingPath)
+	return "", nil
+}
+
+func (m *csiBlockMapper) MapDevice(devicePath, globalMapPath, volumeMapPath, volumeMapName string, podUID types.UID) error {
```

Comment:
We are ignoring supplied `devicePath` parameter in this function and we only use return value of `NodePublish` as real device path. This may or may not be true and is a departure from previous version of this code and I am not sure if it is backward compatible change.  In old version of code - if `NodePublish` returned empty string then value returned by `WaitForAttach` was considered as canonical device path.

Reply:
@gnufied 

Thank you for your coment.

Fixed by taking `devicePath` returned by `SetupDevice`.
(There's no acutal device at `devicePath` by the time that `SetupDevice` returns, however it would be safe.)

Reply:
I don't follow, we don't use devicePath so should we set it to _? A short comment could explain that the plugin doesn't accept a devicePath because it cannot know the devicePath, i.e. whatever path on the node where the device actually exists, until after it has called NodePublish.